src/Features/CustomerEffortScoreTracks.php

```diff
@@ -40,17 +40,44 @@ class CustomerEffortScoreTracks {
 	 */
 	const PRODUCT_UPDATE_ACTION_NAME = 'product_update';
 
+	/**
+	 * Action name for settings change.
+	 */
+	const SETTINGS_CHANGE_ACTION_NAME = 'settings_change';
+
 	/**
 	 * Label for the snackbar that appears when a user submits the survey.
 	 *
 	 * @var string
 	 */
 	private $onsubmit_label;
 
+
 	/**
 	 * Constructor. Sets up filters to hook into WooCommerce.
 	 */
 	public function __construct() {
+		add_action(
+			'admin_init',
+			array(
+				$this,
+				'maybe_clear_ces_tracks_queue',
+			)
+		);
+
+		$this->add_actions();
```

Comment:
Is there a reason why we wouldn't want 'admin_init' to also be setup in `add_actions`? And in particular, only is tracking is allowed?

If we do want to keep this separate, maybe `add_actions` should be renamed something like `maybe_enable_survey_enqueing` or `enable_survey_enqueing_if_tracking_is_enabled`.

As it stands, it feels a little confusing/unorganized that we add an action handler and then call another function that says it too is adding actions.

Reply:
I think we can include admin_init in the add_actions as well. I assumed that we need to run maybe_clear_ces_tracks_queue on every call, but it looks like that's not the case after looking at the code again. In that case, I think we can remove add_actions and move the code back to the constructor. I totally agree that add_actions is not the best name for the method.